torch/_subclasses/meta_utils.py

```diff
@@ -269,6 +269,180 @@ def empty_create(inner_t, inner_src, symbolic_context=symbolic_context):
                 device="meta",
             )
 
+        # Creates a subclass instance with empty inner tensors according to the specified
+        # symbolic context.
+        def empty_create_subclass(
+            t,
+            outer_size,
+            outer_stride,
+            symbolic_context=symbolic_context,
+            callback=callback,
+            source=source,
+        ):
+            from torch._dynamo.source import AttrSource
+            from torch.fx.experimental.symbolic_shapes import SubclassSymbolicContext
+
+            assert symbolic_context is None or isinstance(
+                symbolic_context, SubclassSymbolicContext
+            )
+
+            # Note: transform_subclass will use __tensor_unflatten__ to generate
+            # a fresh subclass wrapper. We assume that if the inner tensors of
+            # the subclass are given symbolic sizes, their sizes will be used
+            # to construct the (symbolic) sizes of the wrapper tensor.
```

Comment:
via symbolic_context, right?

Reply:
Actually I lifted this from below and just realized this comment is just out-of-date now that we're using `outer_size` / `outer_stride`. Removing the second sentence as it's not true anymore.